opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -58,39 +61,100 @@ def __init__(
         self._meter_provider = meter_provider
 
     def create_counter(self, name, unit=None, description=None) -> APICounter:
-        return Counter(self._instrumentation_info, name, unit, description)
+        return Counter(
+            self._instrumentation_info,
+            name,
+            # pylint: disable=protected-access
+            self._meter_provider._measurement_consumer,
+            unit,
+            description,
+        )
 
     def create_up_down_counter(
         self, name, unit=None, description=None
     ) -> APIUpDownCounter:
         return UpDownCounter(
-            self._instrumentation_info, name, unit, description
+            self._instrumentation_info,
+            name,
+            # pylint: disable=protected-access
+            self._meter_provider._measurement_consumer,
+            unit,
+            description,
         )
 
     def create_observable_counter(
         self, name, callback, unit=None, description=None
     ) -> APIObservableCounter:
+
+        # pylint: disable=protected-access
+        (
+            self._meter_provider._measurement_consumer.register_asynchronous_instrument(
+                self
+            )
+        )
```

Comment:
This should be creating the instrument, calling `register_asynchronous_instrument(instrument)`, then returning the instrument

Reply:
Correct, fixing :+1:

Reply:
Fixed